backend/internetarchive/internetarchive.go

```diff
@@ -0,0 +1,1086 @@
+		"x-amz-filemeta-rclone-mtime": swift.TimeToFloatString(modTime),
+
+		// we add some more headers for intuitive actions
+		"x-amz-auto-make-bucket":     "1",    // create an item if does not exist, do nothing if already
+		"x-archive-auto-make-bucket": "1",    // same as above in IAS3 original way
+		"x-archive-keep-old-version": "0",    // do not keep old versions (a.k.a. trashes in other clouds)
+		"x-archive-meta-mediatype":   "data", // mark media type of the uploading file as "data"
+		"x-archive-queue-derive":     "0",    // skip derivation process (e.g. encoding to smaller files, OCR on PDFs)
+		"x-archive-cascade-delete":   "1",    // enable "cascate delete" (delete all derived files in addition to the file itself)
+	}
+	if size >= 0 {
+		headers["Content-Length"] = fmt.Sprintf("%d", size)
+		headers["x-archive-size-hint"] = fmt.Sprintf("%d", size)
+	}
+
+	// read the md5sum if available
+	var md5sumHex string
+	if !o.fs.opt.DisableChecksum {
+		md5sumHex, err = src.Hash(ctx, hash.MD5)
+		if err == nil && matchMd5.MatchString(md5sumHex) {
+			// Set the md5sum in header on the object if
+			// the user wants it
+			// https://github.com/jjjake/internetarchive/blob/245637653/internetarchive/item.py#L969
+			headers["Content-MD5"] = md5sumHex
+		}
+	}
+
+	// make a PUT request at (IAS3)/encoded(:item/:path)
+	var resp *http.Response
+	opts := rest.Opts{
+		Method:        "PUT",
+		Path:          "/" + url.PathEscape(path.Join(bucket, bucketPath)),
+		Body:          in,
+		ContentLength: &size,
+		ExtraHeaders:  headers,
+	}
+
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		return o.fs.shouldRetry(resp, err)
+	})
+
+	// we can't update/find metadata here as IA will "ingest" uploaded file(s)
+	// upon uploads. (you can find its progress at https://archive.org/history/ItemNameHere )
+	// or we have to wait for finish? (needs polling (frontend)/metadata/:item or scraping (frontend)/history/:item)
+	var newObj *Object
+	if err == nil {
+		newObj, err = o.fs.waitFileUpload(ctx, o.remote, o.fs.getHashes(ctx, src), size)
+	} else {
+		newObj = &Object{}
+	}
+	o.crc32 = newObj.crc32
+	o.md5 = newObj.md5
+	o.sha1 = newObj.sha1
+	o.modTime = newObj.modTime
+	o.size = newObj.size
+	return err
+}
+
+// Remove an object
+func (o *Object) Remove(ctx context.Context) (err error) {
+	bucket, bucketPath := o.split()
+
+	// make a DELETE request at (IAS3)/:item/:path
+	var resp *http.Response
+	opts := rest.Opts{
+		Method: "DELETE",
+		Path:   "/" + url.PathEscape(path.Join(bucket, bucketPath)),
+	}
+
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		return o.fs.shouldRetry(resp, err)
+	})
+
+	// deleting files can take bit longer as
+	// it'll be processed on same queue as uploads
+	if err == nil {
+		err = o.fs.waitDelete(ctx, bucket, bucketPath)
+	}
+	return err
+}
+
+// MimeType of an Object if known, "" otherwise
```

Comment:
If this method isn't supported then you don't need to define it, so just remove it.

Reply:
I was totally forgotten it was an optional :sweat_smile: